crates/wash-lib/src/parser/mod.rs

```diff
@@ -510,3 +544,63 @@ impl RawProjectConfig {
         })
     }
 }
+
+impl ProjectConfig {
+    pub fn resolve_registry_url(&self) -> Result<String> {
+        let registry_url = match &self.common.registry.url {
+            Some(url) => url.clone(),
+            None => {
+                bail!("No registry URL specified in wasmcloud.toml");
+            }
+        };
+
+        Ok(registry_url)
+    }
+
+    pub async fn resolve_registry_credentials(
+        &self,
+        registry: impl AsRef<str>,
+    ) -> Result<RegistryAuth> {
+        let credentials_file = &self
+            .common
+            .registry
+            .credentials
+            .to_owned()
+            .or_else(|| default_registry_credentials_file().ok());
+
+        if credentials_file.is_none() {
+            return Ok(RegistryAuth::Anonymous);
+        }
+
+        let credentials_file = credentials_file.clone().unwrap();
```

Comment:
Nit: is this clone necessary? And you can avoid the unwrap by combining this with the above block, like so:
```rust
let Some(credentials_file) = credentials_file else {
  return Ok(RegistryAuth::Anonymous);
};
```

You can repeat this structure a couple times below as well